mailbox/plugin/quota-search-elasticsearch/src/main/java/org/apache/james/quota/search/elasticsearch/events/ElasticSearchQuotaMailboxListener.java

```diff
@@ -64,17 +67,21 @@ public boolean isHandling(Event event) {
     }
 
     @Override
-    public void event(Event event) throws IOException {
-        handleEvent((QuotaUsageUpdatedEvent) event);
+    public Publisher<Void> reactiveEvent(Event event) {
+        try {
+            return handleEvent((QuotaUsageUpdatedEvent) event);
+        } catch (IOException e) {
+            return Mono.error(e);
+        }
```

Comment:
onErrorResume will be shorter?

Reply:
onErrorResume catches errors  evaluations within the pipeline.

Here I'm handling errors while generating the pipeline.